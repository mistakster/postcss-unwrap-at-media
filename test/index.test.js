var fs = require('fs');
var postcss = require('postcss');

var plugin = require('../index');

describe('unwrap @media', function () {

	it('should remove all @media rules', function () {
		var result = postcss([plugin])
			.process(
				'@media (min-width: 720px) {\n    div {\n        color: red;\n    }\n}'
			);
		expect(result.css).toMatchSnapshot();
	});

	it('shouldn\'t touch other rules', function () {
		var result = postcss([plugin])
			.process(
				'p {\n    margin: 1em 0;\n}\n\n@media (min-width: 720px) {\n    div {\n        color: red;\n    }\n}'
			);
		expect(result.css).toMatchSnapshot();
	});

	it('should delete a block with particular @media rules', function () {
		var result = postcss([plugin({ disallow: 'print' })])
			.process(
				'@media print {\n    div {\n        color: red;\n    }\n}\n\n' +
				'@media screen {\n    div {\n        color: red;\n    }\n}\n\n' +
				'p {\n    margin: 1em 0;\n}'
			);
		expect(result.css).toMatchSnapshot();
	});

	it('should delete a block with any of specified @media rules', function () {
		var result = postcss([plugin({ disallow: ['print', 'screen'] })])
			.process(
				'@media print {\n    div {\n        color: red;\n    }\n}\n\n' +
				'@media screen {\n    div {\n        color: red;\n    }\n}\n\n' +
				'p {\n    margin: 1em 0;\n}'
			);
		expect(result.css).toMatchSnapshot();
	});

});
