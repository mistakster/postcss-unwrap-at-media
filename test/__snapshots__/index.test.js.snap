// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unwrap @media should delete a block with @media rules matches the predicate 1`] = `
"div {
        color: black;
    
}

div {
        color: red;
    
}

p {
    margin: 1em 0;
}"
`;

exports[`unwrap @media should delete a block with any of specified @media rules 1`] = `
"p {
    margin: 1em 0;
}"
`;

exports[`unwrap @media should delete a block with particular @media rules 1`] = `
"div {
        color: red;
    
}

p {
    margin: 1em 0;
}"
`;

exports[`unwrap @media should remove all @media rules 1`] = `
"div {
        color: red;
    
}"
`;

exports[`unwrap @media shouldn't touch other rules 1`] = `
"p {
    margin: 1em 0;
}

div {
        color: red;
    
}"
`;
